src/plugins/undoRedo/undoRedo.js

```diff
@@ -49,7 +49,7 @@ function UndoRedo(instance) {
       return;
     }
 
-    var originalData = plugin.instance.getSourceDataArray();
+    var originalData = plugin.instance.getSourceData();
```

Comment:
Storing a reference to source data isn't the best choice.
Maybe you can find a different way (without storing the reference) to save removed data?

Reply:
It already used `getSourceDataArray`, what's the difference?

Reply:
`.getSourceData()` returns reference to source data:
https://github.com/handsontable/handsontable/blob/master/src/core.js#L2085
then
https://github.com/handsontable/handsontable/blob/master/src/dataSource.js#L42
`toArray` is false, so `getData()` return `dataSource.data`

but `getSourceDataArray()` returns a new array with values from source data, there is no reference, or maybe better, shouldn't keep any reference to source data.
https://github.com/handsontable/handsontable/blob/master/src/core.js#L2110
then 
https://github.com/handsontable/handsontable/blob/master/src/dataSource.js#L42
but this time `toArray` is true, so our result is `datasource.getByRange(start, end, toArray)`

I hope my explanation is clear enough. Let me know it isn't.

Reply:
On line 56 it does a `deepClone`, which does a JSON deep-clone; so it does a complete deep copy of the data; so no reference to the initial data is exposed.

Reply:
You're right. I didn't notice hidden part of code on a PR preview.